pages/_app/index.jsx

```diff
@@ -78,41 +80,39 @@ class App extends NextApp {
     }
   }
 
-  fetchUser = async () => {
-    try {
-      const { dataProvider, activity } = new Route(window.location.pathname)
-      await this.store.init(dataProvider, activity)
-      this.setState({ isAuthorized: true })
-      Sentry.configureScope((scope) => {
-        scope.setUser({
-          id: this.store.user.id,
-          email: this.store.user.email,
-        })
-      })
-    } catch (unauthorizedError) {
-      // TODO: Do some check before redirect
-      this.setState({ isAuthorized: false })
-      if (!this.isRouteWithoutStore) this.redirectToLogin()
-      throw unauthorizedError
-    }
-  }
-
   get isRouteWithoutStore() {
     const { router } = this.props
     return ROUTES_WITHOUT_STORE.find((r) => router.asPath.startsWith(r))
   }
 
+  get isAuthorized() {
+    return !!this.store?.user?.id
+  }
+
   async componentDidMount() {
     const { router } = this.props
+    router.events.on('routeChangeStart', this.handleRouteChange)
 
     window.addEventListener('unhandledrejection', this.handlePromiseRejection)
 
-    if (!this.isRouteWithoutStore) await this.fetchUser()
+    if (this.isRouteWithoutStore) return
 
-    // we use routeChangeStart to immediately update selected activity
-    // ensure it's bound after user is fetched to avoid false errors
-    //  - user has no access to data provider
-    router.events.on('routeChangeStart', this.handleRouteChange)
+    try {
+      await this.store.init()
```

Comment:
you need to pass here dataProviderID otherwise pages cannot be accessed directly. http://localhost:3000/data-providers/1/overview

Reply:
It does work for me

Reply:
It works because of `softReload`. It triggers route change and the store gets updated.

Reply:
Shoot sorry. I forgot to pull the latest changes. Nonetheless, it works just because the code is really fast and data provider is changed before next pretender, right?

Reply:
Anyway, I changed it already